indico/MaKaC/PDFinterface/conference.py

```diff
@@ -643,715 +632,742 @@ def _processTOCPage(self):
+            color_cell = " "
         if self._ttPDFFormat.showContribAbstract():
-            spkList=[]
-            for spk in contrib.getSpeakerList():
-                spkName = spk.getFullName()
+            speaker_list = []
+            for spk in contrib.speakers:
+                person = spk.person
+                speaker_name = person.get_full_name(last_name_first=False, show_title=True)
                 if not self._ttPDFFormat.showSpeakerTitle():
-                    spkName = self._getNameWithoutTitle(spk)
-                if self._showSpeakerAffiliation and spk.getAffiliation().strip() != "":
-                    spkName += " (" + spk.getAffiliation() + ")"
-                spkList.append(spkName)
-            if len(spkList) > 0:
-                if len(spkList) == 1:
-                    speakerWord = i18nformat(""" _("Presenter"): """)
+                    speaker_name = self._getNameWithoutTitle(person)
+                if self._showSpeakerAffiliation and person.affiliation:
+                    speaker_name += " ({})".format(person.affiliation)
+                speaker_list.append(speaker_name)
+
+            if speaker_list:
+                if len(speaker_list) == 1:
+                    speaker_title = i18nformat(""" _("Presenter"): """)
                 else:
-                    speakerWord = i18nformat(""" _("Presenters"): """)
-                speakerText = speakerWord + ", ".join(spkList)
-                speakerText = "<font name=\"Times-Italic\"><i>%s</i></font>"%speakerText
-                lt.append([self._fontify(speakerText,9)])
-            captionText=escape(contrib.getDescription())
-            lt.append([self._fontify(captionText,9)])
-            captionAndSpeakers = Table(lt,colWidths=(None),style=self._tsSpk)
-            if colorCell != "":
-                l.append([colorCell,date,captionAndSpeakers])
+                    speaker_title = i18nformat(""" _("Presenters"): """)
+                speaker_content = speaker_title + ", ".join(speaker_list)
+                speaker_content = '<font name="Times-Italic"><i>{}</i></font>'.format(speaker_content)
+                lt.append([self._fontify(speaker_content, 9)])
+            caption = escape(contrib.description)
+            lt.append([self._fontify(caption, 9)])
+            caption_and_speakers = Table(lt, colWidths=None, style=self._tsSpk)
+            if color_cell:
+                l.append([color_cell, date, caption_and_speakers])
             else:
-                l.append([date,captionAndSpeakers])
+                l.append([date, caption_and_speakers])
         else:
-            caption = Table(lt,colWidths=(None),style=self._tsSpk)
-            spkList=[]
-            for spk in contrib.getSpeakerList():
-                spkName = spk.getFullName()
+            caption = Table(lt, colWidths=None, style=self._tsSpk)
+            speaker_list = []
+            for speaker in contrib.speakers:
+                person = speaker.person
```

Comment:
Same `speaker` vs. `person` issue. By the way, this seems remarkably similar to the code we've rewritten just above. Maybe it deserves a function of itself?